src/shared/settings.js

```diff
@@ -46,6 +46,33 @@ function jsonConfigsFrom(document) {
   return config;
 }
 
+/**
+ * Returns a `hypothesisConfig` object from the window, or `{}`.
+ *
+ * Find the `window.hypothesisConfig` object and merge the properties
+ * of it onto the config parameter passed in.
+ *
+ * If the `config` param doesn't exist, instantiate a new object.
+ *
+ * If the `window.hypothesisConfig` exists as a function and returns an
+ * object containing properties, add it to the return object.
+ *
+ * @param {HypothesisConfig} config - an existing config object.
+ * @returns {HypothesisConfig} - an updated config object.
+ */
+function jsonConfigsFromWindow(config) {
```

Comment:
Hmm ... this duplicates some of the logic from `config-func-settings-from.js` but with parts missing. Now that I look into how we are using `jsonConfigsFrom` in `src/annotator/config/settings.js`. I see that there are other cases where we respect configuration provided via JSON `<script>` tags but not equivalent properties from the `window.hypothesisConfig` function. Specifically the "annotations" and "query" properties.

My inclination would be to combine those two sources into one function to avoid this hazard. However, I just noticed another issue related to `window.hypothesisConfig`'s pure-ness.